src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,226 @@
+                + "JOIN cars_drivers cd "
+                + "ON d.id = cd.driver_id "
+                + "WHERE cd.car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, carId);
+            ResultSet resultSet = statement.executeQuery();
+            List<Driver> drivers = new ArrayList<>();
+            while (resultSet.next()) {
+                drivers.add(getDriver(resultSet));
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to retrieve the car drivers"
+                    + " using the car id : . " + carId, e);
+        }
+    }
+
+    private Driver getDriver(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("id", Long.class);
+        String name = resultSet.getString("name");
+        String licenseNumber = resultSet.getString("license_number");
+        return new Driver(id, name, licenseNumber);
+    }
+
+    private static void updateCarFields(Car car) {
+        String query = "UPDATE cars "
+                + "SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("can`t update "
+                    + car + " in carsDB.", e);
+        }
+    }
+
+    private static void deleteCarDriversByCarId(Car car) {
+        String query = "DELETE FROM cars_drivers "
+                + "WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("cant delete drivers from "
+                    + car + "cars_drivers", e);
+        }
+    }
+
+    private static void insertCarDrivers(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) "
+                + "VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            for (Driver driver : car.getDrivers()) {
+                statement.setLong(1, car.getId());
+                statement.setLong(2, driver.getId());
+                statement.addBatch();
+                statement.clearParameters();
```

Comment:
```suggestion
                statement.addBatch();
                statement.clearParameters();
```

why do u need this?

Reply:
It seems like I got a bit too caught up with it and confused myself.... 🙈  To avoid executing each query individually, I add them to a batch using addBatch(). Then, I use executeBatch() to execute the entire batch as a single query to the database. I just realized that I don't need to call clearParameters() and addBatch() inside the loop. I only need to call addBatch() once inside the loop for each query, and then use executeBatch() outside the loop to execute the entire batch.